sympy/core/basic.py

```diff
@@ -1586,7 +1586,11 @@ def _eval_rewrite(self, pattern, rule, **hints):
 
         if pattern is None or isinstance(self, pattern):
             if hasattr(self, rule):
-                rewritten = getattr(self, rule)(*args)
+                hints.pop('deep', None)
+                if hints:
+                    rewritten = getattr(self, rule)(*args, **hints)
```

Comment:
I have assumed that those `_eval_rewrite_as_xxx` methods that have no `**hints` parameter declared would not accept `hints` even if it is empty, but maybe I was mistaken. It would be interesting to see if this single line could be used in all cases. Then the alternate original version with not hints could be omitted.